src/components/Proposal/Predictions/PredictionButtons.tsx

```diff
@@ -227,46 +230,95 @@ class PredictionBox extends React.Component<IProps, IState> {
-            secondaryHeader={"> " + formatTokens(proposal.upstakeNeededToPreBoost, "GEN") + " for boost!"}
-          /> : ""
-        }
-
-        <div className={css.stakeControls}>
-          {stakingEnabled
-            ? <span>
-              {
-                (!this.props.currentAccountAddress ? "" : tip(VoteOptions.No) !== "") ?
-                  <Tooltip placement="left" trigger={["hover"]} overlay={tip(VoteOptions.No)}>
-                    {passButton}
-                  </Tooltip> :
-                  passButton
-              }
-              {
-                (!this.props.currentAccountAddress ? "" : tip(VoteOptions.Yes) !== "") ?
-                  <Tooltip placement="left" trigger={["hover"]} overlay={tip(VoteOptions.Yes)}>
-                    {failButton}
-                  </Tooltip> :
-                  failButton
-              }
-            </span>
-            : <span className={css.disabledPredictions}>
-              Predictions are disabled
-            </span>
+    if (contextMenu) {
+      return (
+        <div className={wrapperClass}>
+          {showPreStakeModal ?
+            <PreTransactionModal
+              actionType={pendingPrediction === VoteOptions.Yes ? ActionTypes.StakePass : ActionTypes.StakeFail}
+              action={(amount: number) => { stakeProposal(proposal.dao.address, proposal.id, pendingPrediction, amount); }}
+              beneficiaryProfile={beneficiaryProfile}
+              closeAction={this.closePreStakeModal.bind(this)}
+              currentAccountGens={currentAccountGens}
+              dao={dao}
+              proposal={proposal}
+              secondaryHeader={"> " + formatTokens(proposal.upstakeNeededToPreBoost, "GEN") + " for boost!"}
+            /> : ""
           }
+          <div className={css.contextTitle}>
+            <div>
+              <span>
+                Predict
+              </span>
+            </div>
+          </div>
+          <div className={css.contextContent}>
+            {stakingEnabled
+              ? <div>
+                {
+                  (!this.props.currentAccountAddress ? "" : tip(VoteOptions.No) !== "") ?
+                    <Tooltip placement="left" trigger={["hover"]} overlay={tip(VoteOptions.No)}>
+                      {passButton}
+                    </Tooltip> :
+                    passButton
+                }
+                {
+                  (!this.props.currentAccountAddress ? "" : tip(VoteOptions.Yes) !== "") ?
+                    <Tooltip placement="left" trigger={["hover"]} overlay={tip(VoteOptions.Yes)}>
+                      {failButton}
+                    </Tooltip> :
+                    failButton
+                }
+              </div>
+              : <span className={css.disabledPredictions}>
+                Predictions are disabled
+              </span>
+            }
+          </div>
         </div>
-      </div>
-    );
+      );
+    } else {
+      return (
+        <div className={wrapperClass}>
```

Comment:
this looks very duplicated from above, do we really need to duplicate all this code? what actually changes for the "context menu"?

Reply:
The proposal card for mobile still uses the other way of voting.. Not sure a more elegant way to do it. Perhaps a mobile design for the new cards will allow us to do away with the duplication? @A-Zak